src/app/inventory/store/stats.ts

```diff
@@ -413,12 +419,37 @@ function buildPlugStats(
 function buildLiveStats(
   stats: DestinyItemStatsComponent,
   itemDef: DestinyInventoryItemDefinition,
+  createdItem: DimItem,
   defs: D2ManifestDefinitions,
   statGroup: DestinyStatGroupDefinition,
   statDisplays: { [key: number]: DestinyStatDisplayDefinition }
 ) {
   const ret: DimStat[] = [];
 
+  // Sum all the conditionally inactive and active plug stats from sockets so we can calculate
+  // the value and base. The live stat includes all mod stats whether they are active or not.
```

Comment:
I don't believe this is the case! I think the live stats show the "correct" stats, so if the conditional stat isn't active, it won't be included.

Reply:
I'm 99% sure they are included in the live stats but I will get some evidence.

Reply:
this is NOT what i expected, but i think ryan is right.
my live *character* stats are correct but the live *armor* stats seem to assume the Powerful Friends +20 perk is active even when it isn't.

this is so counterintuitive, and i am so worried about caching, i would love to find a third person to check it

Reply:
I haven't had a chance to sit down and collect evidence but I see the stats come through on both charged harvester and powerful friends even when the condition shouldn't be active. It would be good to get a third person to check it.

Reply:
Ok so evidence inbound.

An interesting thing to note is that you can see the issue with negative mods on stats in the charged harvester example. One of the stats is 10 when it should be 8.

## Powerful Friends
### In game
![image](https://user-images.githubusercontent.com/7344652/103499767-594d4400-4e9d-11eb-8b9a-b6d1eb65277a.png)

### Beta
![image](https://user-images.githubusercontent.com/7344652/103499791-6d914100-4e9d-11eb-99fb-a5852700902b.png)

### This branch
![image](https://user-images.githubusercontent.com/7344652/103499811-826dd480-4e9d-11eb-85b7-27dad4113469.png)

#### Stats Object on the item 
```json
[
    {
      "investmentValue": 26,
      "statHash": 2996146975,
      "displayProperties": {
        "description": "Increases your movement speed and maximum jump height.",
        "name": "Mobility",
        "icon": "/common/destiny2_content/icons/c9aa8439fc71c9ee336ba713535569ad.png",
        "hasIcon": true
      },
      "sort": 22,
      "value": 6,
      "base": 6,
      "statMayBeWrong": false,
      "maximumValue": 42,
      "bar": true,
      "smallerIsBetter": false,
      "additive": true
    },
    {
      "investmentValue": 10,
      "statHash": 392767087,
      "displayProperties": {
        "description": "Increases the amount of damage you can take before dying.",
        "name": "Resilience",
        "icon": "/common/destiny2_content/icons/9f5f65d08b24defb660cebdfd7bae727.png",
        "hasIcon": true
      },
      "sort": 23,
      "value": 10,
      "base": 10,
      "statMayBeWrong": false,
      "maximumValue": 42,
      "bar": true,
      "smallerIsBetter": false,
      "additive": true
    },
    {
      "investmentValue": 16,
      "statHash": 1943323491,
      "displayProperties": {
        "description": "Increases the speed at which you regain lost health.",
        "name": "Recovery",
        "icon": "/common/destiny2_content/icons/47e16a27c8387243dcf9b5d94e26ccc4.png",
        "hasIcon": true
      },
      "sort": 24,
      "value": 16,
      "base": 16,
      "statMayBeWrong": false,
      "maximumValue": 42,
      "bar": true,
      "smallerIsBetter": false,
      "additive": true
    },
    {
      "investmentValue": 7,
      "statHash": 1735777505,
      "displayProperties": {
        "description": "Discipline decreases the cooldown time of your grenades, allowing you to use them more often.",
        "name": "Discipline",
        "icon": "/common/destiny2_content/icons/ca62128071dc254fe75891211b98b237.png",
        "hasIcon": true
      },
      "sort": 25,
      "value": 7,
      "base": 7,
      "statMayBeWrong": false,
      "maximumValue": 42,
      "bar": true,
      "smallerIsBetter": false,
      "additive": true
    },
    {
      "investmentValue": 15,
      "statHash": 144602215,
      "displayProperties": {
        "description": "Intellect decreases the cooldown of your Super ability, allowing you to use it more often.",
        "name": "Intellect",
        "icon": "/common/destiny2_content/icons/59732534ce7060dba681d1ba84c055a6.png",
        "hasIcon": true
      },
      "sort": 26,
      "value": 15,
      "base": 15,
      "statMayBeWrong": false,
      "maximumValue": 42,
      "bar": true,
      "smallerIsBetter": false,
      "additive": true
    },
    {
      "investmentValue": 7,
      "statHash": 4244567218,
      "displayProperties": {
        "description": "Strength decreases the cooldown time of your melee ability, allowing you to use it more often.",
        "name": "Strength",
        "icon": "/common/destiny2_content/icons/c7eefc8abbaa586eeab79e962a79d6ad.png",
        "hasIcon": true
      },
      "sort": 27,
      "value": 7,
      "base": 7,
      "statMayBeWrong": false,
      "maximumValue": 42,
      "bar": true,
      "smallerIsBetter": false,
      "additive": true
    },
    {
      "investmentValue": 61,
      "statHash": -1000,
      "displayProperties": { "name": "Total" },
      "sort": 28,
      "value": 61,
      "base": 61,
      "statMayBeWrong": false,
      "maximumValue": 100,
      "bar": false,
      "smallerIsBetter": false,
      "additive": false
    },
    {
      "investmentValue": 44,
      "statHash": -1100,
      "displayProperties": {
        "name": "Custom Total",
        "description": "Custom total of selected base stats, ignoring mods or masterworks. Visit Settings to configure which stats are included."
      },
      "sort": 29,
      "value": 44,
      "base": 44,
      "statMayBeWrong": false,
      "maximumValue": 100,
      "bar": false,
      "smallerIsBetter": false,
      "additive": false
    }
  ]
```

## Charge Harvester
### In game
![image](https://user-images.githubusercontent.com/7344652/103499899-d11b6e80-4e9d-11eb-92fe-db8226f71263.png)

### Beta
![image](https://user-images.githubusercontent.com/7344652/103499939-f4deb480-4e9d-11eb-9298-bb0a61b05682.png)

### This branch
![image](https://user-images.githubusercontent.com/7344652/103499962-0f189280-4e9e-11eb-9fa9-0619dbc6c808.png)

#### Stats object on item
```json
[
    {
      "investmentValue": 0,
      "statHash": 2996146975,
      "displayProperties": {
        "description": "Increases your movement speed and maximum jump height.",
        "name": "Mobility",
        "icon": "/common/destiny2_content/icons/c9aa8439fc71c9ee336ba713535569ad.png",
        "hasIcon": true
      },
      "sort": 22,
      "value": 0,
      "base": 10,
      "statMayBeWrong": true,
      "maximumValue": 42,
      "bar": true,
      "smallerIsBetter": false,
      "additive": true
    },
    {
      "investmentValue": 0,
      "statHash": 392767087,
      "displayProperties": {
        "description": "Increases the amount of damage you can take before dying.",
        "name": "Resilience",
        "icon": "/common/destiny2_content/icons/9f5f65d08b24defb660cebdfd7bae727.png",
        "hasIcon": true
      },
      "sort": 23,
      "value": 10,
      "base": 10,
      "statMayBeWrong": true,
      "maximumValue": 42,
      "bar": true,
      "smallerIsBetter": false,
      "additive": true
    },
    {
      "investmentValue": 3,
      "statHash": 1943323491,
      "displayProperties": {
        "description": "Increases the speed at which you regain lost health.",
        "name": "Recovery",
        "icon": "/common/destiny2_content/icons/47e16a27c8387243dcf9b5d94e26ccc4.png",
        "hasIcon": true
      },
      "sort": 24,
      "value": 13,
      "base": 13,
      "statMayBeWrong": false,
      "maximumValue": 42,
      "bar": true,
      "smallerIsBetter": false,
      "additive": true
    },
    {
      "investmentValue": 10,
      "statHash": 1735777505,
      "displayProperties": {
        "description": "Discipline decreases the cooldown time of your grenades, allowing you to use them more often.",
        "name": "Discipline",
        "icon": "/common/destiny2_content/icons/ca62128071dc254fe75891211b98b237.png",
        "hasIcon": true
      },
      "sort": 25,
      "value": 10,
      "base": 10,
      "statMayBeWrong": false,
      "maximumValue": 42,
      "bar": true,
      "smallerIsBetter": false,
      "additive": true
    },
    {
      "investmentValue": 6,
      "statHash": 144602215,
      "displayProperties": {
        "description": "Intellect decreases the cooldown of your Super ability, allowing you to use it more often.",
        "name": "Intellect",
        "icon": "/common/destiny2_content/icons/59732534ce7060dba681d1ba84c055a6.png",
        "hasIcon": true
      },
      "sort": 26,
      "value": 6,
      "base": 6,
      "statMayBeWrong": false,
      "maximumValue": 42,
      "bar": true,
      "smallerIsBetter": false,
      "additive": true
    },
    {
      "investmentValue": 14,
      "statHash": 4244567218,
      "displayProperties": {
        "description": "Strength decreases the cooldown time of your melee ability, allowing you to use it more often.",
        "name": "Strength",
        "icon": "/common/destiny2_content/icons/c7eefc8abbaa586eeab79e962a79d6ad.png",
        "hasIcon": true
      },
      "sort": 27,
      "value": 14,
      "base": 14,
      "statMayBeWrong": false,
      "maximumValue": 42,
      "bar": true,
      "smallerIsBetter": false,
      "additive": true
    },
    {
      "investmentValue": 53,
      "statHash": -1000,
      "displayProperties": { "name": "Total" },
      "sort": 28,
      "value": 53,
      "base": 63,
      "statMayBeWrong": true,
      "maximumValue": 100,
      "bar": false,
      "smallerIsBetter": false,
      "additive": false
    },
    {
      "investmentValue": 39,
      "statHash": -1100,
      "displayProperties": {
        "name": "Custom Total",
        "description": "Custom total of selected base stats, ignoring mods or masterworks. Visit Settings to configure which stats are included."
      },
      "sort": 29,
      "value": 39,
      "base": 39,
      "statMayBeWrong": true,
      "maximumValue": 100,
      "bar": false,
      "smallerIsBetter": false,
      "additive": false
    }
  ]
```

## Radiant Light
### In game
![image](https://user-images.githubusercontent.com/7344652/103500027-46873f00-4e9e-11eb-9f4a-119407002748.png)

### Beta
![image](https://user-images.githubusercontent.com/7344652/103500044-54d55b00-4e9e-11eb-8c17-02958cd013ef.png)

### This branch
![image](https://user-images.githubusercontent.com/7344652/103500057-5d2d9600-4e9e-11eb-88ea-9b1206aafb25.png)

#### Stat object on item
```json
[
    {
      "investmentValue": 7,
      "statHash": 2996146975,
      "displayProperties": {
        "description": "Increases your movement speed and maximum jump height.",
        "name": "Mobility",
        "icon": "/common/destiny2_content/icons/c9aa8439fc71c9ee336ba713535569ad.png",
        "hasIcon": true
      },
      "sort": 22,
      "value": 7,
      "base": 7,
      "statMayBeWrong": false,
      "maximumValue": 42,
      "bar": true,
      "smallerIsBetter": false,
      "additive": true
    },
    {
      "investmentValue": 10,
      "statHash": 392767087,
      "displayProperties": {
        "description": "Increases the amount of damage you can take before dying.",
        "name": "Resilience",
        "icon": "/common/destiny2_content/icons/9f5f65d08b24defb660cebdfd7bae727.png",
        "hasIcon": true
      },
      "sort": 23,
      "value": 10,
      "base": 10,
      "statMayBeWrong": false,
      "maximumValue": 42,
      "bar": true,
      "smallerIsBetter": false,
      "additive": true
    },
    {
      "investmentValue": 13,
      "statHash": 1943323491,
      "displayProperties": {
        "description": "Increases the speed at which you regain lost health.",
        "name": "Recovery",
        "icon": "/common/destiny2_content/icons/47e16a27c8387243dcf9b5d94e26ccc4.png",
        "hasIcon": true
      },
      "sort": 24,
      "value": 13,
      "base": 13,
      "statMayBeWrong": false,
      "maximumValue": 42,
      "bar": true,
      "smallerIsBetter": false,
      "additive": true
    },
    {
      "investmentValue": 12,
      "statHash": 1735777505,
      "displayProperties": {
        "description": "Discipline decreases the cooldown time of your grenades, allowing you to use them more often.",
        "name": "Discipline",
        "icon": "/common/destiny2_content/icons/ca62128071dc254fe75891211b98b237.png",
        "hasIcon": true
      },
      "sort": 25,
      "value": 12,
      "base": 12,
      "statMayBeWrong": false,
      "maximumValue": 42,
      "bar": true,
      "smallerIsBetter": false,
      "additive": true
    },
    {
      "investmentValue": 11,
      "statHash": 144602215,
      "displayProperties": {
        "description": "Intellect decreases the cooldown of your Super ability, allowing you to use it more often.",
        "name": "Intellect",
        "icon": "/common/destiny2_content/icons/59732534ce7060dba681d1ba84c055a6.png",
        "hasIcon": true
      },
      "sort": 26,
      "value": 11,
      "base": 11,
      "statMayBeWrong": false,
      "maximumValue": 42,
      "bar": true,
      "smallerIsBetter": false,
      "additive": true
    },
    {
      "investmentValue": 27,
      "statHash": 4244567218,
      "displayProperties": {
        "description": "Strength decreases the cooldown time of your melee ability, allowing you to use it more often.",
        "name": "Strength",
        "icon": "/common/destiny2_content/icons/c7eefc8abbaa586eeab79e962a79d6ad.png",
        "hasIcon": true
      },
      "sort": 27,
      "value": 7,
      "base": 7,
      "statMayBeWrong": false,
      "maximumValue": 42,
      "bar": true,
      "smallerIsBetter": false,
      "additive": true
    },
    {
      "investmentValue": 60,
      "statHash": -1000,
      "displayProperties": { "name": "Total" },
      "sort": 28,
      "value": 60,
      "base": 60,
      "statMayBeWrong": false,
      "maximumValue": 100,
      "bar": false,
      "smallerIsBetter": false,
      "additive": false
    },
    {
      "investmentValue": 43,
      "statHash": -1100,
      "displayProperties": {
        "name": "Custom Total",
        "description": "Custom total of selected base stats, ignoring mods or masterworks. Visit Settings to configure which stats are included."
      },
      "sort": 29,
      "value": 43,
      "base": 43,
      "statMayBeWrong": false,
      "maximumValue": 100,
      "bar": false,
      "smallerIsBetter": false,
      "additive": false
    }
  ]
```

Reply:
Well that's just bonkers. So we actually have to subtract from the live stats when they're not enabled. SMH.

Reply:
Yeah it's crazy isn't it. I wonder if its actually an API bug?